docs/spec_draft.md

```diff
@@ -341,6 +351,174 @@ only exists to establish data dependencies from `result` to `inputs`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.all_gather
+
+### Semantics
+
+For each replica subgroup in `replica_groups`, the operation
+[concatenates](#stablehloconcatenate) the replicas of `input` tensor formed
+using the subgroup along `all_gather_dim` dimension following the order of the
+replicas in the subgroup and produces a `result` tensor.
+
+`channel_handle`, an optional input, is a handle given to a user to represent a
+communication channel between two computations via a
+[stablehlo.send](#stablehlosend) and [stablehlo.recv](#stablehlorecv)
+instruction pair or via collective instructions (e.g.
+[stablehlo.all_reduce](#stablehloall_reduce),
+[stablehlo.all_gather](#stablehloall_gather), etc.). It is represented using a
+`channel-id` and a `channel-type`. A `channel-id` is used for cross-partition
+communication; only operations with the same opcode and `channel-id` can
+communicate to each other. Non-positive `channel-id` is equivalent to no channel
+id. The `channel-type` is not relevant for this instruction.
+
+In the instruction syntax, multiple replica ids are grouped as a replica
+subgroup and multiple replica subgroups form `replica_groups`. If
+`use_global_device_ids = true`, the ids in `replica_groups` are interpreted as
+global device ids given by `rid * num_pids + pid` where `num_pids` $=$ number of
+partitions.
+
+The execution of the operation involves logically partitioning each execution
+instance (`ei`) of `rid`, `pid` pair, henceforth referred to as `ei(rid, pid)`
+into groups. All the execution instances within a group can participate in the
+operation without getting interfered by execution instances from other groups.
+The number of groups represents the number of `stablehlo.all_gather` operations
+executed in parallel.
```

Comment:
Does this mean that one `all_gather` with `replica_groups = [[0, 2], [1, 3]]` produces the same two tensors as two `all_gather` operations, one with  `replica_groups = [[0, 2]]` and one with  `replica_groups = [[1, 3]]` (and otherwise similar arguments)?